pageserver/src/tenant.rs

```diff
@@ -2834,13 +2882,6 @@ impl Tenant {
             lsn
         });
 
-        // Create a placeholder for the new branch. This will error
-        // out if the new timeline ID is already in use.
-        let timeline_uninit_mark = {
-            let timelines = self.timelines.lock().unwrap();
-            self.create_timeline_uninit_mark(dst_id, &timelines)?
-        };
-
```

Comment:
Now the lock order with respect to `gc_cs` changes, and historically there has been a lot or long issues about races related to creating timelines with "out of scope" (gc cutoff) LSNs... I do remember trying to understand what could had been the issue from the current state, but did not really discover anything interesting.

Did you consider the `gc_cs` here?

Reply:
I think the comment starting `// First acquire the GC lock so that another task cannot advance the GC` might need a bit of tuneup as well, because we are clearly no longer starting by doing this as we already have timelineuninitmark.

Reply:
I think we're okay here: the `gc_cs` is an async lock whereas self.timelines is a std::sync lock, so the order would always have been to lock `Tenant::timelines` within `gc_cs` and not vice versa.

The checks still look right to me: we're now checking for duplicate TimelineId at the very start of the request, then holding an effective lock on that timeline ID in timelines_creating, and during that span we're then acquring gc_cs, then validating LSNs. 

The comment is indeed a bit confusing, updated it in f59564588

Reply:
Oki.

Reply:
Went over as well, looks fine.